developer/src/kmc-ldml/src/compiler/tran.ts

```diff
@@ -26,6 +27,18 @@ class TransformCompiler<T extends TransformCompilerType, TranBase extends Tran>
     super(source, callbacks);
   }
 
+  // TODO-LDML: dup with 'vars'
```

Comment:
now vars and tran each initialize their own KmnCompiler (see other comment).  Seems excessive but?

Note that the parser is used by callers of the sections, not just within a compiler.

Reply:
TODO: refactor the multiple instances of KmnCompiler into the top-level ldml compiler and pass as parameter. This also avoids async pollution down into the depths of the compiler.

Reply:
@mcdurdin  maybe in callbacks?

Reply:
Let's discuss, I don't think it needs to be outside kmc-ldml. Having a single module which is responsible for instantiation and lifecycle of the KmnCompiler is enough, within kmc-ldml. It should be a lazy initializing singleton.

Reply:
@mcdurdin going to do that refactor in a followon PR ok?